Userland/Libraries/LibCore/Stream.h

```diff
@@ -527,11 +527,9 @@ class BufferedHelper {
         if (!stream.is_open())
             return ENOTCONN;
 
-        auto maybe_buffer = ByteBuffer::create_uninitialized(buffer_size);
-        if (!maybe_buffer.has_value())
-            return ENOMEM;
+        auto buffer = TRY(ByteBuffer::create_uninitialized(buffer_size));
 
-        return BufferedType<T> { move(stream), maybe_buffer.release_value() };
+        return BufferedType<T> { move(stream), buffer };
```

Comment:
Move this buffer in.
I'm pretty sure the copy ctor actually duplicates the buffer